src/Illuminate/Routing/ResourceRegistrar.php

```diff
@@ -53,7 +53,7 @@ public function register($name, $controller, array $options = [])
         // We need to extract the base resource from the resource name. Nested resources
         // are supported in the framework, but we need to know what name to use for a
         // place-holder on the route wildcards, which should be the base resources.
-        $base = $this->getResourceWildcard(last(explode('.', $name)));
+        $base = str_singular($this->getResourceWildcard(last(explode('.', $name))));
```

Comment:
Please don't use the global function and use the Str class like everywhere else in the framework core.